engine/src/main/java/org/apache/apex/engine/plugin/DefaultApexPluginDispatcher.java

```diff
@@ -87,18 +87,24 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor)
     };
 
     executorService = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
-        blockingQueue, new NameableThreadFactory("PluginExecutorThread"), rejectionHandler);
+        blockingQueue, new NameableThreadFactory("PluginExecutorThread", true), rejectionHandler);
   }
 
   @Override
   protected void serviceStop() throws Exception
   {
-    executorService.shutdownNow();
-    executorService.awaitTermination(10, TimeUnit.SECONDS);
-    if (!executorService.isTerminated()) {
-      LOG.warn("Executor service still active for plugins");
+    if (executorService != null) {
```

Comment:
@vrozov interesting that AbstractService calls for this. Normally I would be leery about adding additional code that promotes bad behavior, if we start defending methods like this it quickly becomes a slippery slope. In this case it is ok as the parent is calling out for it.